### PR TITLE
Support regions in Geobuf file format

### PR DESCRIPTION
**Pull request by FredrikNysjo**
Geobuf typically compresses GeoJSON files by 50%, and is a lossless format, so can be nice to have support for it in TissUUmaps.

### CODE REVIEW

File: tissuumaps/static/js/utils/regionUtils.js [hunk 1]
```diff
@@ -1798,24 +1798,44 @@ regionUtils.JSONToRegions= function(filepath){
...
             filepath = path + "/" + filepath;
         }
-        fetch(filepath)
-        .then((response) => {
-            return response.json();
-        })
-        .then((regionsobj) => {
-            regionUtils.JSONValToRegions(regionsobj);
-        });
+        if (filepath.includes(".pbf")) {
```
Comment (jwindhager) [inline tissuumaps/static/js/utils/regionUtils.js:1801]: Is there any reason for not checking with `.endsWith`? Also, how about uppercase?

File: tissuumaps/static/js/utils/regionUtils.js [hunk 2]
```diff
@@ -1798,24 +1798,44 @@ regionUtils.JSONToRegions= function(filepath){
...
+            // Load GeoJSON stored in Geobuf format (https://github.com/mapbox/geobuf)
+            fetch(filepath)
+            .then((response) => {
+                return response.arrayBuffer();
+            })
+            .then((regionsobj) => {
+                const data = new Pbf(regionsobj);
+                regionUtils.JSONValToRegions(geobuf.decode(data));
+            });
```
Comment (jwindhager) [inline tissuumaps/static/js/utils/regionUtils.js:1810]: Why not combine the two `then`s?

File: tissuumaps/static/js/utils/regionUtils.js [hunk 3]
```diff
@@ -1798,24 +1798,44 @@ regionUtils.JSONToRegions= function(filepath){
...
+        } else if (filepath.includes(".geojson") || filepath.includes(".json")) {
```
Comment (jwindhager) [inline tissuumaps/static/js/utils/regionUtils.js:1811]: Same as above (endsWith, case)

File: tissuumaps/static/js/utils/regionUtils.js [hunk 4]
```diff
@@ -1798,24 +1798,44 @@ regionUtils.JSONToRegions= function(filepath){
...
+            fetch(filepath)
+            .then((response) => {
+                return response.json();
+            })
+            .then((regionsobj) => {
+                regionUtils.JSONValToRegions(regionsobj);
+            });
```
Comment (jwindhager) [inline tissuumaps/static/js/utils/regionUtils.js:1818]: Same as above (combine)

File: tissuumaps/static/js/utils/regionUtils.js [hunk 5]
```diff
@@ -1798,24 +1798,44 @@ regionUtils.JSONToRegions= function(filepath){
...
+                const data = new Pbf(event.target.result);
+                regionUtils.JSONValToRegions(geobuf.decode(data));
+            };
+            reader.readAsArrayBuffer(file);
+        } else if (file.name.includes(".geojson") || file.name.includes(".json")) {
+            reader.onload=function(event) {
+                regionUtils.JSONValToRegions(JSON.parse(event.target.result));
+            };
+            reader.readAsText(file);
+        }
```
Comment (jwindhager) [inline tissuumaps/static/js/utils/regionUtils.js:1838]: Would be good to have an `else` clause here, warning of unsupported file types

**Review by jwindhager — CHANGES_REQUESTED** [review]

File: tissuumaps/static/js/utils/regionUtils.js (hunk 1 above)
Comment (FredrikNysjo) [reply tissuumaps/static/js/utils/regionUtils.js]: No. Fixed it now, and also added conversion to lowercase.

File: tissuumaps/static/js/utils/regionUtils.js (hunk 5 above)
Comment (FredrikNysjo) [reply tissuumaps/static/js/utils/regionUtils.js:1846]: Thanks, added it now!

File: tissuumaps/static/js/utils/regionUtils.js (hunk 1 above)
Comment (cavenel) [reply tissuumaps/static/js/utils/regionUtils.js]: > Also, how about uppercase?

That's interesting, I don't think the flask code supports uppercase extensions. Should we fix this everywhere? Can be quite a lot of changes.

Comment (cavenel) [reply tissuumaps/static/js/utils/regionUtils.js]: After thinking about it a bit, let's not go there yet. We support lowercase file extensions for now within the GUI.
We keep this for later, as flask is url based and difference between lowercase/uppercase urls can be tricky to handle in some cases.

**Review by cavenel — APPROVED** [review]

File: tissuumaps/static/js/utils/regionUtils.js (hunk 2 above)
Comment (jwindhager) [reply tissuumaps/static/js/utils/regionUtils.js]: Ah, didn't realize that `arrayBuffer` is `async`, too. My mistake!

**Review by jwindhager — APPROVED** [review]

File: tissuumaps/static/js/utils/regionUtils.js (hunk 2 above)
Comment (FredrikNysjo) [reply tissuumaps/static/js/utils/regionUtils.js]: Yes, it only returns a promise, so both were needed.

**Pull request merged**